.github/workflows/deploy-network.yml

```diff
@@ -0,0 +1,201 @@
+        run: echo "retrieve_state_output=$(aws s3 cp s3://${{ secrets.AWS_S3_BLADE_BUCKET }}/states/${{ inputs.environment }} state.json)" >> $GITHUB_OUTPUT
+      - name: Check state file
+        id: check_state_file
+        if: contains(steps.retrieve_state.outputs.retrieve_state_output, 'download')
+        run: echo "resources=$(cat state.json | jq -r ".resources" | jq length)" >> $GITHUB_OUTPUT
+      - name: Set RPC URL
+        id: rpc_url
+        if: contains(steps.retrieve_state.outputs.retrieve_state_output, 'download')
+        run: echo "url=$(cat state.json | jq -r '.outputs.aws_lb_ext_domain.value // empty')" >> $GITHUB_OUTPUT
+  deploy_network:
+    name: Deploy the network
+    runs-on: ubuntu-latest
+    needs: check_network
+    if: needs.check_network.outputs.check_output == 0
+    environment: ${{ inputs.environment }}
+    outputs:
+      terraform_output: ${{ steps.terraform_failure.outputs.terraform_output }}
+      ansible_output: ${{ steps.ansible_failure.outputs.ansible_output }}
+      rpc_url: ${{ steps.rpc_url.outputs.url }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.1.1
+        with:
+          repository: Ethernal-Tech/blade-deployment
+          ref: main
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4.0.1
+        with:
+          aws-region: ${{ vars.AWS_REGION }}
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+      - name: Install Terraform
+        uses: hashicorp/setup-terraform@v3.0.0
+        with:
+          terraform_version: 1.4.5
+      - name: Configure Terraform
+        run: sed 's/# backend "s3" {}/backend "s3" {}/' main.tf > main.tf.tmp && mv main.tf.tmp main.tf
+      - name: Terraform Init
+        run: terraform init -backend-config="bucket=${{ secrets.AWS_S3_BLADE_BUCKET }}" -backend-config="key=states/${{ inputs.environment }}" -backend-config="region=${{ vars.AWS_REGION }}"
+      - name: Terraform Validate
+        run: terraform validate -no-color
+        continue-on-error: true
+      - name: Terraform Apply
+        run: terraform apply -auto-approve
+        env:
+          TF_VAR_deployment_name: ${{ inputs.environment }}
+          TF_VAR_base_instance_type: ${{ vars.AWS_INSTANCE_TYPE }}
+          TF_VAR_geth_count: ${{ vars.GETH_COUNT }}
+          TF_VAR_fullnode_count: ${{ vars.FULLNODE_COUNT }}
+          TF_VAR_validator_count: ${{ vars.VALIDATOR_COUNT }}
+      - name: Terraform Failed
+        if: failure()
+        id: terraform_failure
+        run: echo "terraform_output=false" >> $GITHUB_OUTPUT
+      - name: Configure private keys
+        run: |
+          terraform output pk_ansible > ~/private.key
+          chmod 600 ~/private.key 
+          eval "$(ssh-agent)"
+          ssh-add ~/private.key
+      - name: Install Ansible / botocore / boto3
+        run: |
+          python3 -m pip install --user ansible
+          python3 -m pip install boto3 botocore
```

Comment:
Asking just out of curiosity:
are `boto3` and `botocore` dependencies required by the `ansible`?

Reply:
They are not mandatory dependencies for `ansible`, but we require them for AWS-related tasks utilized within `ansible playbooks`.